### PR TITLE
feat: add option cwd

### PR DESCRIPTION
**Pull request by mojoaxel**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -67,11 +67,11 @@
   },
   "devDependencies": {
     "grunt": "^1.0.1",
-    "grunt-auto-release": "^0.0.6",
+    "grunt-auto-release": "^0.0.7",
```
Comment (eddiemonge) [inline package.json]: can you move these to a new PR?

File: tasks/bump.js [hunk 2]
```diff
@@ -30,6 +33,20 @@ module.exports = function(grunt) {
       versionType: false
     });
 
+    // check if cwd exists
+    if (!fs.existsSync(opts.cwd)) {
+      grunt.fatal('The in "cwd" defined directory does not exist');
```
Comment (eddiemonge) [inline tasks/bump.js]: `The "cwd" requested does not exist`

File: tasks/bump.js [hunk 3]
```diff
@@ -30,6 +33,20 @@ module.exports = function(grunt) {
...
+    }
+
+    // set cwd as the working directory for exec
+    var execOpts = {
+      cwd: path.resolve(opts.cwd)
+    };
+
+    // set grunts default working directory
+    grunt.verbose.writeln('Changing working directory: ' + opts.cwd);
```
Comment (eddiemonge) [inline tasks/bump.js]: Only show this if cwd `!== '.'`

File: tasks/bump.js [hunk 4]
```diff
@@ -30,6 +33,20 @@ module.exports = function(grunt) {
...
+    grunt.file.setBase(opts.cwd);
```
Comment (eddiemonge) [inline tasks/bump.js]: I don't know if this sets this just for this task or all tasks so maybe store the current base and then restore it after the tasks have run.

**Review by eddiemonge — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (mojoaxel) [inline package.json]: yeah. ok. it's cleaner that way.

File: tasks/bump.js (hunk 2 above)
Comment (mojoaxel) [inline tasks/bump.js]: :+1:

File: tasks/bump.js (hunk 3 above)
Comment (mojoaxel) [inline tasks/bump.js]: I think it doesn't hurt if the current working directory is shown to the user. This is only in `--verbose` mode. But I'm also ok, with the change.

File: tasks/bump.js (hunk 4 above)
Comment (mojoaxel) [inline tasks/bump.js]: Good point! I haven't tested it with multiple tasks. I better change it back!

File: tasks/bump.js [hunk 5]
```diff
@@ -60,6 +81,11 @@ module.exports = function(grunt) {
     var next = function() {
       if (!queue.length) {
         grunt.config.set('bump.version', globalVersion);
+
+        // reset original cwd
+        if (originalCwd) {
+          grunt.file.setBase(originalCwd);
```
Comment (eddiemonge) [inline tasks/bump.js]: wont this mess up the file list for every exec after the first? Should the original changeBase go in the runIf?

**Comment by eddiemonge** [comment]
Closing due to inactivity

**Pull request closed without merging**